li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/common/LargeMessageHeaderValue.java

```diff
@@ -108,4 +116,22 @@ public static LargeMessageHeaderValue fromBytes(byte[] bytes) {
     }
     return new LargeMessageHeaderValue(type, new UUID(mostSignificantBits, leastSignificantBits), segmentNumber, numberOfSegments, INVALID_MESSAGE_SIZE);
   }
+
+  /**
+   * Check if the value of this segment header is valid
+   * @return the check result
+   */
+  public boolean isValid() {
+    if (_type > LargeMessageSegment.CURRENT_VERSION) {
```

Comment:
Can I change the LargeMessageSegment.CURRENT_VERSION to LargeMessageHeaderValue.V3 for the compatibility test?
the context is: In this test, I will pass messages of V3 version to the consumer. But such messages can not pass the sanity check of the LargeMessageHeaderValue here as our LargeMessageSegment.CURRENT_VERSION is smaller than V3.

Reply:
> Can I change the LargeMessageSegment.CURRENT_VERSION to LargeMessageHeaderValue.V3 for the compatibility test? 

No. You can't change it just for the sake of testing. You should see where the public variable `CURRENT_VERSION` is being used. if you change it directly in the class, then you are breaking the `DefaultSegmentSerializer`, which will now add `V3` in its segment header encoding. 

I think a better way to handle this to add a test only interface that will allow you to set the current version in the LargeMessageSegment or in the LargeMessageHeaderValue. It depends on how you are testing the compatibility.

Reply:
Yeah, I agree with your words that I can not change the value of LargeMessageSegment.CURRENT_VERSION to V3. What I want to express is that changing `if (_type > LargeMessageSegment.CURRENT_VERSION)` to `if (_type > LargeMessageHeaderValue.V3)`. It is not elegant and somehow breaks the integrity of the code. That's why I ask it.
Your advice about adding a new interface is pretty useful. But my worry is that when adding such a setter, our LargeMessageSegment.CURRENT_VERSION will not be a final constant, which makes me feel a little odd. Is this change(i.e. make LargeMessageSegment.CURRENT_VERSION not final) acceptable?

Reply:
Remove this redundant check